File: rust_backend/src/service/wms_service.rs
use crate::repository::wms_repository::WmsRepository;
use reqwest::{
    header::{HeaderMap, AUTHORIZATION},
    Client, Error, Response, StatusCode,
};
use std::collections::HashMap;

pub struct WmsService<R: WmsRepository> {
    repository: R,
}

impl<R: WmsRepository> WmsService<R> {
    pub fn new(repository: R) -> Self {
        WmsService { repository }
    }

    pub async fn forward_wms_request(
        &self,
        id: u32,
        query_params: HashMap<String, String>,
    ) -> Result<Response, Error> {
        let wms = match self.repository.get_by_id(id) {
            Some(wms) => wms,
            None => {}
        };
    }
}

File: backend/internal/service/wms_service.go
package service

import (
	"backend/internal/model"
	"backend/internal/repository"
)

type WmsService struct {
	repo repository.WmsRepository
}

func NewWmsService(repo repository.WmsRepository) *WmsService {
	return &WmsService{repo: repo}
}

func (s *WmsService) GetAll() ([]model.Wms, error) {
	return s.repo.GetAll()
}

func (s *WmsService) GetById(id int) (*model.Wms, error) {
	return s.repo.GetById(id)
}
